language/move-lang/src/parser/lexer.rs

```diff
@@ -317,6 +317,28 @@ fn find_token(file: &'static str, text: &str, start_offset: usize) -> Result<(To
                     )]);
                 }
                 (Tok::ByteStringValue, 2 + len + 1)
+            } else if text.starts_with("b\"") {
+                let mut last_quote = 1;
+                loop {
+                    match text[last_quote + 1..].find('"') {
+                        Some(index) => {
+                            last_quote += index;
+                        }
+                        None => {
+                            return Err(vec![(
+                                make_loc(file, start_offset, start_offset + 2),
+                                "Missing closing quote (\") after byte string".to_string(),
+                            )]);
+                        }
+                    }
+
+                    if text[last_quote..].starts_with("\\\"") {
+                        last_quote += 1;
+                    } else {
+                        break;
+                    }
+                }
```

Comment:
I think this can be done with a filter 
```
let last_quote = text[2..]
    .enumerate()
    .filter(|(idx, c)| c == '"' && !text[idx - 1..].starts_with("\\"))
    .map(|(idx, _)| idx)
    .next()
    .ok_or_else(|| {
        Err(vec![(
            make_loc(file, start_offset, start_offset + 2),
            "Missing closing quote (\") after byte string".to_string(),
        )])
    })?;
```
But I'll leave it to @bob-wilson for final style here

Reply:
The more I think about it, I think this won't really work at all for escape sequences. Let me take a deeper look

Reply:
Yes. with a filter, it looks prettier.